### PR TITLE
handle prometheus alerts

### PR DESCRIPTION
**Pull request by yosefmih**
handles prometheus alerts

### CODE REVIEW

File: api/server/handlers/webhook/prometheus_incoming.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	})
+	label_key_values := ""
+	for _, alert := range prometheusAlert.Alerts {
+		for k, v := range alert.Labels {
+			label_key_values += fmt.Sprintf("%s %s", k, v)
+		}
+		if alert.Labels["alertname"] == "NoopAlert" {
+			continue
+		}
+		desiredReplicas, err := strconv.ParseInt(alert.Labels["desiredReplicas"], 10, 32)
```
Comment (ferozemohideen) [inline api/server/handlers/webhook/prometheus_incoming.go:86]: can we decode the Labels map into a struct?

File: api/server/handlers/webhook/prometheus_incoming.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	recordPrometheusAlertRequest := connect.NewRequest(&porterv1.RecordPrometheusAlertRequest{
+		ProjectId: projectId,
+		ClusterId: clusterId,
+	})
+	label_key_values := ""
+	for _, alert := range prometheusAlert.Alerts {
+		for k, v := range alert.Labels {
+			label_key_values += fmt.Sprintf("%s %s", k, v)
+		}
+		if alert.Labels["alertname"] == "NoopAlert" {
```
Comment (ferozemohideen) [inline api/server/handlers/webhook/prometheus_incoming.go:83]: why does the request make it to the server if it is a no-op?

File: api/server/handlers/webhook/prometheus_incoming.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+}
+
+func (p *PrometheusAlertWebhookHandler) handlePrometheusAlert(ctx context.Context, projectId, clusterId int64, prometheusAlert *types.PrometheusAlert) error {
+	ctx, span := telemetry.NewSpan(ctx, "porter-process-prom-alert")
+	defer span.End()
+	recordPrometheusAlertRequest := connect.NewRequest(&porterv1.RecordPrometheusAlertRequest{
+		ProjectId: projectId,
+		ClusterId: clusterId,
+	})
+	label_key_values := ""
```
Comment (ferozemohideen) [inline api/server/handlers/webhook/prometheus_incoming.go:78]: prefer camelCase for this

**Review by ferozemohideen — APPROVED** [review]

File: api/server/handlers/webhook/prometheus_incoming.go (hunk 2 above)
Comment (yosefmih) [reply api/server/handlers/webhook/prometheus_incoming.go:83]: We use that as a heartbeat

File: api/server/handlers/webhook/prometheus_incoming.go (hunk 1 above)
Comment (yosefmih) [reply api/server/handlers/webhook/prometheus_incoming.go:86]: Hmm I will look into that, actually we are not setting the replica labels right now, removing this part of the code

Comment (yosefmih) [reply api/server/handlers/webhook/prometheus_incoming.go:86]: But we basically have map[string]string so a struct would simply obfuscate that atm

**Pull request merged**